model/postgres/postgres_server.rb

```diff
@@ -136,6 +140,14 @@ def needs_recycling?
     vm.display_size != resource.target_vm_size || storage_size_gib != resource.target_storage_size_gib
   end
 
+  def replica_lsn_caught_up
```

Comment:
This function can be used for standbys as well, no need to prefix with `replica`.

Actually, I think we should use this in wait_catch_up label for standbys as well. That way we wouldn't need different logic for nearly same functionality.

Reply:
I'm not sure if that's a good idea because I think the logic for standby is better. The reason is, it only queries the primary. For read replicas, the lag is calculated by querying both of the servers. If you think that's a fair trade-off, and want to have a single logic, I can still change it, though. However, I believe it's better to keep it this way.

Reply:
`wait_catch_up` label only gets executed for few hours max and during that time, I think it is OK to run query in both places.

That said, I believe, the recovering server does not accept any queries until some point. Does that affect us here? Though. it would affect both standby and read replicas. IIRC, not needing to deal that case was one of my reasons to only query primary in the past.